source/adaptive_load/plugin_util.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
```

Comment:
Can we try to find a more specific name for this library? Using a generic name like *util* will eventually attract any code related to plugins. How about plugin_loader.h?

Feel free to choose any other name that is specific enough not to attract unrelated code.

Reply:
Done